core/src/color.rs

```diff
@@ -33,11 +39,40 @@ impl Color {
         a: 0.0,
     };
 
+    /// New Color with range checks
+    pub fn new(r: f32, g: f32, b: f32, a: f32) -> Color {
+        debug_assert!(
+            (0.0f32..=1.0f32).contains(&r),
+            "Red component must be on [0, 1]"
+        );
+        debug_assert!(
+            (0.0f32..=1.0f32).contains(&g),
+            "Green component must be on [0, 1]"
+        );
+        debug_assert!(
+            (0.0f32..=1.0f32).contains(&b),
+            "Blue component must be on [0, 1]"
+        );
+        debug_assert!(
+            (0.0f32..=1.0f32).contains(&a),
```

Comment:
I believe the compiler should be able to infer the floating-point precision here, so we may be able to remove the `f32` suffixes.

Reply:
Fixed in daf1c639afaba989e8c4551a01fe0058ef9d1824